File: offer/22.链表中倒数第 k 个节点.go
type ListNode struct {
	Val  int
	Next *ListNode
}

func getKthFromEnd(head *ListNode, k int) *ListNode {
	fast := head
	slow := head
	for i := 0; i < k; i++ {
		if fast == nil {
			return nil // 如果链表长度小于 k，返回 nil
		}
		fast = fast.Next
	}
	for fast != nil {
		fast = fast.Next
		slow = slow.Next
	}
	return slow
}

File: offer/22.链表中倒数第k个节点.js
function getKthFromEnd(head, k) {
  let fast = head;
  let slow = head;
  for (let i = 0; i < k; i++) {
    if (fast === null) {
      return null; // 如果链表长度小于 k，返回 null
    }
    fast = fast.next;
  }
  while (fast !== null) {
    fast = fast.next;
    slow = slow.next;
  }
  return slow;
}